experimental/framework/frs-client/src/frsClient.ts

```diff
@@ -0,0 +1,157 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { Container, Loader } from "@fluidframework/container-loader";
+import {
+    ContainerSchema,
+    DOProviderContainerRuntimeFactory,
+    FluidContainer,
+    RootDataObject,
+} from "@fluid-experimental/fluid-static";
+
+import { IRuntimeFactory } from "@fluidframework/container-definitions";
+import { RouterliciousService } from "@fluid-experimental/get-container";
+
+import {
+    FRSContainerConfig,
+} from "./interfaces";
+
+/**
+ * FRSClientInstance provides the ability to have a Fluid object backed by a FRS service
+ */
+export class FRSClientInstance {
+    constructor(private readonly serviceConnectionConfig: RouterliciousService) {
+    }
+
+    public async createContainer(
+        serviceContainerConfig: FRSContainerConfig,
+        containerSchema: ContainerSchema,
+    ): Promise<FluidContainer> {
+        const runtimeFactory = new DOProviderContainerRuntimeFactory(
+            containerSchema,
+        );
+        const container = await this.getContainerCore(
+            serviceContainerConfig,
+            runtimeFactory,
+            true,
+        );
+
+        const rootDataObject = await this.getRootDataObject(container);
+        return new FluidContainer(container, rootDataObject);
+    }
+
+    public async getContainer(
+        serviceContainerConfig: FRSContainerConfig,
+        containerSchema: ContainerSchema,
+    ): Promise<FluidContainer> {
+        const runtimeFactory = new DOProviderContainerRuntimeFactory(
+            containerSchema,
+        );
+        const container = await this.getContainerCore(
+            serviceContainerConfig,
+            runtimeFactory,
+            false,
+        );
+
+        const rootDataObject = await this.getRootDataObject(container);
+        return new FluidContainer(container, rootDataObject);
+    }
+
+    private async getRootDataObject(
+        container: Container,
+    ): Promise<RootDataObject> {
+        const rootDataObject = (await container.request({ url: "/" })).value;
```

Comment:
Please use requestFluidObject<RootDataObject>() that does a bit more checking here.

Reply:
@vladsud if we had a Fluid-specific linter, would this be a pattern we could check for? I've been kicking around the idea of some custom lint rules, but I'm still hunting for enough patterns to make it worthwhile.

Reply:
For sure if it's smart (i.e. could do complex AST analyzes). I.e. we detect that caller either does not check response.mimeType at all (like here), or checks it to be "fluid/object".
Maybe if it's dumb - in this case it would be correct 99% of the time, but might be annoying if I expect json return value, or had some custom error code checking.